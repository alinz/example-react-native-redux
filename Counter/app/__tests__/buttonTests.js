/**
 * Created by rotem on 27/08/2016.
 */
import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../components/button';


describe('Button', () => {
    it('renders correctly', () => {
        const tree = renderer.create(
            <Button/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders correctly with text', () => {
        const tree = renderer.create(
            <Button text="Up"/>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});