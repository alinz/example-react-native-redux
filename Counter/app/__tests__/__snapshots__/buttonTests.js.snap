exports[`Button renders correctly 1`] = `
<View
  accessibilityLabel="button undefined"
  accessible={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "lightgray",
      "height": 30,
      "justifyContent": "center",
      "margin": 3,
      "padding": 10,
      "width": 100
    }
  }
  testID="button undefined">
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function bound touchableHandleResponderGrant]}
    onResponderMove={[Function bound touchableHandleResponderMove]}
    onResponderRelease={[Function bound touchableHandleResponderRelease]}
    onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
    onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
    onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
    style={
      Object {
        "opacity": 1
      }
    }
    testID={undefined}>
    <Text />
  </View>
</View>
`;

exports[`Button renders correctly with text 1`] = `
<View
  accessibilityLabel="button Up"
  accessible={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "lightgray",
      "height": 30,
      "justifyContent": "center",
      "margin": 3,
      "padding": 10,
      "width": 100
    }
  }
  testID="button Up">
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function bound touchableHandleResponderGrant]}
    onResponderMove={[Function bound touchableHandleResponderMove]}
    onResponderRelease={[Function bound touchableHandleResponderRelease]}
    onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
    onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
    onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
    style={
      Object {
        "opacity": 1
      }
    }
    testID={undefined}>
    <Text>
      Up
    </Text>
  </View>
</View>
`;
